cpu/load_balancer.py

```diff
@@ -0,0 +1,169 @@
+                /tmp/mpstat.log'
+        process.run(cmd, shell=True)
+        with open("/tmp/mpstat.log", "r") as file:
+            lines = file.readlines()
+        filtered_lines = [line.strip().split() for line in lines if re.match(
+                    "^[0-9][0-9]:[0-9][0-9]:[0-9][0-9]", line)]
+        cpu_list = []
+        usr_list = []
+        for columns in filtered_lines:
+            cpu_list.append(columns[2])
+            usr_list.append(columns[3])
+
+        if len(cpu_list) == on_cpu_count:
+            data = {
+                    'CPU': cpu_list,
+                    '%USER': usr_list,
+                    }
+            df = pd.DataFrame(data)
+            self.log.info("\n %s", df)
+            count = 0
+            for utilization in usr_list:
+                if int(float(utilization)) == int(self.capacity) or \
+                        int(float(utilization)) >= (int(self.capacity) - 2):
+                    count = count + 1
+            if on_cpu_count == count:
+                self.log.info("Load -balancer balnced load across \
+                        the available cpu for smt mode \
+                        %s core's: %s ", smt_mode, core)
+            else:
+                self.fail("Load -balancer is failed to balance load across \
+                        the available cpu for smt mode \
+                        %s core's: %s ", smt_mode, core)
+
+    def test(self):
+        """
+        In this funtion basically we are online and offline the
+        cores and cpu's in sequence,
+        1.Running the stress-ng workload.
+        2.changing the SMT modes.
+        3.Validating the mpstat command output for CPU utilization
+        as per SMT mode.
+        """
+        process.run('ppc64_cpu --cores-on=all', shell=True)
+        process.run('ppc64_cpu --smt=on', shell=True)
+        totalcpus = int(multiprocessing.cpu_count())
+        total_cores = totalcpus//8
+        if (self.no_threads == ""):
+            self.no_threads = totalcpus + 1
+        self.log.info("Total no of cores %d", total_cores)
+        self.log.info("Total no of online cores %d", totalcpus)
+        self.run_workload(self.no_threads, self.cpu_cycles, self.capacity)
+        cpu_controller = ["2", "4", "6", "on", "off"]
+        for core in range(1, total_cores+1):
+            cmd = "ppc64_cpu --cores-on=%s" % (core)
+            self.log.info("Total no of online core's %d", core)
+            process.run(cmd, shell=True)
+            for smt_mode in cpu_controller:
+                cmd = "ppc64_cpu --smt={}".format(smt_mode)
+                self.log.info("smt mode %s", smt_mode)
+                process.run(cmd, shell=True)
+                time.sleep(10)
+                on_cpu_count = cpu.online_count()
+                self.log.info("After SMT mode %s no of \
+                        online cpu's %d", smt_mode, on_cpu_count)
+                self.mpstat_analyzer(on_cpu_count, smt_mode, core)
```

Comment:
- For every core 
  - For every smt mode {off, 2, 4, 6, 8}
    - **Can we save the mpstat log file in the log directory for this testcase? For later analysis.**

Reply:
Presently, `mpstat_analyzer` isn't saving mpstat file